engine/util/static_map.hpp

```diff
@@ -119,24 +108,24 @@ class static_set_base {
   using iterator        = pointer;
   using const_iterator  = pointer;
 
-  constexpr static_set_base(const array<value_type, N>& items)
+  constexpr static_set_base(const std::array<value_type, N>& items)
     : data_(sorted(items))
   {
-    util::span<const T, N> data(data_.begin(), N);
+    util::span<const T, N> data(data_.data(), N);
     assert( is_unique(data) );
   }
 
   constexpr iterator begin() const { return cbegin(); }
-  constexpr iterator cbegin() const { return data_.begin(); }
+  constexpr iterator cbegin() const { return data_.data(); }
 
   constexpr iterator end() const { return cend(); }
-  constexpr iterator cend() const { return data_.end(); }
+  constexpr iterator cend() const { return data_.data() + N; }
```

Comment:
Is there a reason to use data() and data()+N here instead of directly using the corresponding std::array functions?

Reply:
Yeah, different types.

`static_xxx::iterator` is a pointer. `std::array<>::iterator` is implementation defined, just has to satisfy some named requirements.
So while it [`std::array<>::iterator`] is *typically* a pointer, at least in release builds, in msvc debug builds it's a checked iterator struct.

I actually had it call the corresponding `std::array` functions initially, but had to change it (https://github.com/simulationcraft/simc/compare/212a90813c8a97d550e6799fbda4c08e073c3a44..66029fa2b3ec5140a0219d2bfb35a60efda44b0a) because of that.